### PR TITLE
MSPB-107: Set media  control for custom conferece greeting

### PR DESCRIPTION
**Pull request by masmerino13**

### CODE REVIEW

File: submodules/strategy/strategy.js [hunk 1]
```diff
@@ -1400,83 +1400,44 @@ define(function(require) {
...
 
+						monster.pub('common.mediaSelect.render', {
+							container: greetingTemplate.find('.media-wrapper'),
+							selectedOption: _.get(greetingMedia, 'id', null),
+							skin: 'tabs',
+							enableTTS: true,
+							tts: {
+								name: 'MainConferenceGreeting',
+								type: 'conferencingCallInNumber'
+							},
```
Comment (azefiel) [inline submodules/strategy/strategy.js:1411]: Looks like we know which `entity` will be tied to our TTS media here as well: a conference.

**Comment by masmerino13** [comment]
Hi @azefiel,

Hmm do you have more feedback about that bug?

I'm just testing and seems it works with tts and non-tts medias (selected from the dropdown or uploaded)

File: submodules/strategy/strategy.js (hunk 1 above)
Comment (masmerino13) [reply submodules/strategy/strategy.js:1411]: I assume it is, since those are the values that Smart PBX already store when choose a media ;)

**Comment by azefiel** [comment]
voip-main-conference-mediaSelect
Looks like when silence/stream URL is selected, the control does not set it properly in the dropdown.

File: submodules/strategy/strategy.js (hunk 1 above)
Comment (azefiel) [reply submodules/strategy/strategy.js:1411]: Why didn't you set the `entity` prop then?

**Comment by masmerino13** [comment]
I will take a look, thanks!

**Comment by masmerino13** [comment]
Hi @azefiel , I did those fixes/improvements, thanks in advance

**Comment by masmerino13** [comment]
Hi @azefiel, I have updated this PR with the new mediaSelect feature, to enforce end users to select a valid media ID

**Comment by masmerino13** [comment]
Thanks @azefiel !

**Pull request merged**